### ISSUE TITLE
Ticketmaster UK and IE

### ISSUE THREAD

**Issue by allywaring** [mfloto/tmn#1]
Hi I was wondering if I could use this repo for Ticketmaster UK notifications as well. Thanks!

**Comment by mfloto** [comment]
Hey,
thank you for your interest in the project. I just had a look at Ticketmaster UK and unfortunately they don't just use a different endpoint but also structure their data differently. So currently, this project **does not support Ticketmaster UK**.

### The following would need to be done to support Ticketmaster UK:
- [ ] If the supplied country code is `UK`, use the endpoint `ticketmaster.co.uk/api/quickpicks/{EVENT_ID}` in the `get_resale_offers()` function
- [ ] Iterate through all pages (max 20 items per page)
- [ ] New response structs (_Note: the struct for a single offer needs to have many optional fields as the list returned by the api contains multiple different object types. Maybe there is a better solution to deserialize this (custom deserializer, etc.)?_)
- [ ] Filter tickets, as the endpoint returns both standard and resale tickets

### Notes:
- Ticketmaster UK blocks the user-agent `PostmanRuntime/7.32.2`
- It should be possible to set the query `primary` to false. That way, we only get resale offers

**Comment by niallobr** [comment]
Similar issue for Ireland on ticketmaster.ie

```
Run with RUST_BACKTRACE=full to include source snippets.
Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Suggestion: Check if ticketmaster is up or blocked your IP
Could not get ticketmaster response
Unable to get resale offers
Error:
Error: 404
```

Also getting a 404 in web browser when I visit

`https://availability.ticketmaster.eu/api/v2/TM_IE/resale/{EVENT_ID}`

I get a 400 error if I change the country code to NL so it seems like we're in a similar situation to the UK?

I wouldn't be surprised if the .ie website/api was based on the .co.uk version.

Any info @mfloto on how to get up and running would be great.

**Comment by mfloto** [comment]
Hey,
thank you for testing on ticketmaster.ie. The endpoint that seems to work here would be `https://www.ticketmaster.ie/api/quickpicks/{EVENT_ID}/list?sort=price&offset=0&qty=2&primary=false&resale=true&defaultToOne=true` (no cookie, etc. required). It might even be possible to reuse the structs from Ticketmaster UK, as both endpoints look similar.

Currently I don't have time to implement that, but it shouldn't be to difficult.

**Comment by Georgie-Goodhew** [comment]
Hey, is there any update on this? I would like to use it but I have very limited knowledge of coding so I don’t know what anything means. If it is not completed, somebody could explain in simpler terms what needs to be done and how to do it, I could try to help.
